### PR TITLE
session 14

### PR DESCRIPTION
**Pull request by adiliso**

### CODE REVIEW

File: adil/src/Module02/Session14/DataAccess/ProductDao.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+public class ProductDao implements Dao<Product> {
+    private static List<Product> products = new ArrayList<>();
+
+    @Override
+    public void save(Product product) {
+        products.add(product);
+    }
+
+    public List<Product> list() {
```
Comment (sabuhihuseynov) [inline adil/src/Module02/Session14/DataAccess/ProductDao.java:17]: pls use verb for method naming

File: adil/src/Module02/Session14/DataAccess/TypeDao.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class TypeDao implements Dao<Type> {
+    private static List<Type> types = new ArrayList<>();
+
+    @Override
+    public void save(Type type) {
+        try {
+            Type oldType = getByName(type.getName());
```
Comment (sabuhihuseynov) [inline adil/src/Module02/Session14/DataAccess/TypeDao.java:15]: Method name is save but logic inside is update

File: adil/src/Module02/Session14/Entities/Type.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package Module02.Session14.Entities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Type {
+    private String name;
+    private String categoryName;
```
Comment (sabuhihuseynov) [inline adil/src/Module02/Session14/Entities/Type.java:8]: why you don't use Category object here?

File: adil/src/Module02/Session14/Business/ProductManager.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        boolean hasStock = sc.next().equals("Y");
+        sc.nextLine();
+        String type;
+        Type productType;
+        while (true) {
+            System.out.print("Enter product type: ");
+            type = sc.nextLine().toLowerCase();
+            try {
+                productType = typeDao.getByName(type);
+                if (!productType.isActive()) {
```
Comment (sabuhihuseynov) [inline adil/src/Module02/Session14/Business/ProductManager.java:36]:
```
if (productType.isActive()) {
                    break;
                }
                System.out.println("Type is not active. Try again.");
```
                How about this one? I think it is more readable

**Review by sabuhihuseynov — CHANGES_REQUESTED** [review]
pls look

File: adil/src/Module02/Session14/Business/ProductManager.java (hunk 4 above)
Comment (adiliso) [reply adil/src/Module02/Session14/Business/ProductManager.java:36]: nice

File: adil/src/Module02/Session14/DataAccess/TypeDao.java (hunk 2 above)
Comment (adiliso) [reply adil/src/Module02/Session14/DataAccess/TypeDao.java:15]: Yes. But this method can do both operations. I don't need to write update method. I tried to compare it to Spring Boot method.

File: adil/src/Module02/Session14/Entities/Type.java (hunk 3 above)
Comment (adiliso) [reply adil/src/Module02/Session14/Entities/Type.java:8]: onu etsem relationlari qura bilmeyecem e mence. Cetin olacaq yeni

**Review by sabuhihuseynov — APPROVED** [review]

**Pull request merged**